contracts/FastBreak.cdc

```diff
@@ -0,0 +1,682 @@
+        pub fun borrowFastBreakNFT(id: UInt64): &FastBreak.NFT? {
+            post {
+                (result == nil) || (result?.id == id):
+                    "Cannot borrow Fast Break NFT reference: The ID of the returned reference is incorrect"
+            }
+        }
+    }
+
+    /// Capabilities of Fast Break Players
+    ///
+    pub resource interface FastBreakPlayer {
+        pub fun play(
+            fastBreakGameID: String,
+            topShots: [UInt64]
+        ): @FastBreak.NFT
+    }
+
+    /// Fast Break game collection
+    ///
+    pub resource Collection:
+        NonFungibleToken.Provider,
+        NonFungibleToken.Receiver,
+        NonFungibleToken.CollectionPublic,
+        FastBreakNFTCollectionPublic,
+        FastBreakPlayer
+    {
+
+        pub var ownedNFTs: @{UInt64: NonFungibleToken.NFT}
+        pub var numMinted: UInt64
+
+        pub fun withdraw(withdrawID: UInt64): @NonFungibleToken.NFT {
+            let token <- self.ownedNFTs.remove(key: withdrawID) ?? panic("Could not find a fast break with the given ID in the Fast Break collection")
+
+            emit Withdraw(id: token.id, from: self.owner?.address)
+
+            return <-token
+        }
+
+        pub fun deposit(token: @NonFungibleToken.NFT) {
+            let token <- token as! @FastBreak.NFT
+            let id: UInt64 = token.id
+
+            let oldToken <- self.ownedNFTs[id] <- token
+
+            emit Deposit(id: id, to: self.owner?.address)
+
+            destroy oldToken
+        }
+
+        pub fun batchDeposit(tokens: @NonFungibleToken.Collection) {
+            let keys = tokens.getIDs()
+
+            for key in keys {
+                self.deposit(token: <-tokens.withdraw(withdrawID: key))
+            }
+
+            destroy tokens
+        }
+
+        pub fun getIDs(): [UInt64] {
+            return self.ownedNFTs.keys
+        }
+
+        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        }
+
+        pub fun borrowNFTSafe(id: UInt64): &NonFungibleToken.NFT? {
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)
+        }
+
+        pub fun borrowFastBreakNFT(id: UInt64): &FastBreak.NFT? {
+            if self.ownedNFTs[id] != nil {
+                let ref = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+                return ref as! &FastBreak.NFT
+            } else {
+                return nil
+            }
+        }
+
+        /// Play the game of Fast Break with an array of Top Shots
+        ///
+        pub fun play(
+            fastBreakGameID: String,
+            topShots: [UInt64]
+        ): @FastBreak.NFT {
+            pre {
+                FastBreak.fastBreakGameByID.containsKey(fastBreakGameID): "no such fast break game"
+                FastBreak.fastBreakGameByID[fastBreakGameID]!.isPublic: "fast break game is private"
+            }
+
+            let acct = getAccount(self.owner?.address!)
```

Comment:
Not sure if this can really be a constant because the owner can change if the resource is moved to a different account and there wouldn't be a way to update it

Reply:
I see 👍 So depending on which variant is correct, either the initial owner keeps on being used, or always the current owner at each step of the function, maybe make it explicit that the code has that intent.

Reply:
Added context and intention into the comments of this function. Thanks.